### PR TITLE
Close subscription when debug service is stopped.

### PR DESCRIPTION
**Pull request by pisimulation**
Addressed #622

### CODE REVIEW

File: dwds/lib/src/services/debug_service.dart [hunk 1]
```diff
@@ -64,10 +64,18 @@ Future<void> _handleSseConnections(
...
     var responseController = StreamController<Map<String, Object>>();
-    var sub = responseController.stream.map((response) {
+    StreamSubscription sub;
+    sub = responseController.stream.map((response) {
       if (onResponse != null) onResponse(response);
       return jsonEncode(response);
-    }).listen(connection.sink.add);
+    }).listen((response) {
+      unawaited(
+          chromeProxyService.remoteDebugger.onClose.first.whenComplete(() {
```
Comment (natebosch) [inline dwds/lib/src/services/debug_service.dart]: do we need to do this on every event? Or should we do this once?

Comment (pisimulation) [reply dwds/lib/src/services/debug_service.dart]: Just once. Fixed below.

File: dwds/lib/src/services/debug_service.dart [hunk 2]
```diff
@@ -64,10 +64,15 @@ Future<void> _handleSseConnections(
   while (await handler.connections.hasNext) {
     var connection = await handler.connections.next;
     var responseController = StreamController<Map<String, Object>>();
-    var sub = responseController.stream.map((response) {
+    StreamSubscription sub;
+    sub = responseController.stream.map((response) {
       if (onResponse != null) onResponse(response);
       return jsonEncode(response);
     }).listen(connection.sink.add);
+    unawaited(chromeProxyService.remoteDebugger.onClose.first.whenComplete(() {
```
Comment (grouma) [inline dwds/lib/src/services/debug_service.dart]: Does this need to be in the listen? Can this be pulled out?

Comment (pisimulation) [reply dwds/lib/src/services/debug_service.dart]: Done. See new changes.

Comment (grouma) [reply dwds/lib/src/services/debug_service.dart]: I don't see the changes reflected. Also if you pull it out you won't need to have `StreamSubscription sub` you can just do `var sub`

Comment (pisimulation) [reply dwds/lib/src/services/debug_service.dart]: Fixed below

**Pull request merged**